### ISSUE TITLE
"DateTransform" can not be overridden

### ISSUE THREAD

**Issue by keygx** [Hearst-DD/ObjectMapper#609]
ObjectMapper 2.0.0 in Xcode8 and Swift3

public func transformFromJSON(_ value: Any?) -> Date?
public func transformToJSON(_ value: Date?) -> Double?

For access modifier is "public", it can not be overridden.

```
override func transformFromJSON(_ value: Any?) -> Date? {
    // code
}
```
->  _Overriding non-open instance method outside of its defining module_

**Comment by devxoul** [comment]
Duplicated with #580 and fixed in #605.

**Comment by keygx** [comment]
Thanks.
